articles/contributing/code.md

```diff
@@ -0,0 +1,100 @@
+author: cgranade
+ms.author: chgranad
+ms.date: 10/12/2018
+ms.topic: article
+uid: microsoft.quantum.contributing.code
+---
+
+# Contributing Code #
+
+In addition to reporting issues and improving documentation, contributing code to the Quantum Development Kit can be a very direct way to help your peers in the quantum programming community.
+By contributing code, you can help to fix issues, provide new examples, make existing libraries easier to use, or even add entirely new features.
+
+In this guide, we'll detail a bit of what we look for when we review pull requests to help your contribution do the most good.
+
+## What We Look For ##
+
+An ideal code contribution builds on the existing work in a Quantum Development Kit repository to fix issues, expand existing features, or to add new features that are within the scope of a repository.
+When we accept a code contribution, it becomes a part of the Quantum Development Kit itself, such that new features will be released, maintained, and developed in the same way as the rest of the Quantum Development Kit.
+Thus, it is helpful when functionality added by a contribution is well-tested and is documented.
+
+### Unit Tests ###
+
+The Q# functions, operations, and user-defined types that make up libraries such as the canon are automatically tested as a part of development on the [**Microsoft/QuantumLibraries**](https://github.com/Microsoft/QuantumLibraries/) repository.
+When a new pull request is opened, for instance, our [Azure Pipelines](https://azure.microsoft.com/services/devops/pipelines/) configuration will check that the changes in the pull request do not break any existing functionality that the quantum programming community depends on.
+These tests are written using the [Microsoft.Quantum.Xunit](https://www.nuget.org/packages/Microsoft.Quantum.Xunit/) package, which exposes Q# functions and operations as tests for the [xUnit](https://xunit.github.io/) framework.
+
+The [`Canon/tests/Canon.Tests.csproj`](https://github.com/Microsoft/QuantumLibraries/blob/master/Canon/tests/Canon.Tests.csproj) uses this xUnit integration to run any functions or operations ending in `Test`.
+For instance, the following function is used to ensure that the <xref:microsoft.quantum.canon.fst> and <xref:microsoft.quantum.canon.snd> functions both return the right outputs in a representative example.
+If the output of `Fst` or `Snd` is incorrect, the `fail` statement is used to cause the test to fail.
+
+```qsharp
+function PairTest () : Unit {
+    let pair = (12, PauliZ);
+
+    if (Fst(pair) != 12) {
+        let actual = Fst(pair);
+        fail $"Expected 12, actual {actual}.";
+    }
+
+    if (Snd(pair) != PauliZ) {
+        let actual = Snd(pair);
+        fail $"Expected PauliZ, actual {actual}.";
+    }
+}
+```
+
+More complicated conditions can be checked using the techniques in the [testing section](xref:microsoft.quantum.libraries.standard.testing) of the standard libraries guide.
```

Comment:
I'm not sure if this xref changed in 0.6, might it be `microsoft.quantum.libraries.diagnostics` now, pointing to the content at https://github.com/MicrosoftDocs/quantum-docs-pr/blob/master/articles/libraries/standard/diagnostics.md? I can't find the testing section in the libraries any longer...
```suggestion
More complicated conditions can be checked using the techniques in the [testing section](xref:microsoft.quantum.libraries.diagnostics) of the standard libraries guide.
```

Reply:
Good catch, thanks. I don't think that was changed in 0.6 such that there's a pretty good chance I just had the wrong UID in the first place.